tests/metadata/test_merging.py

```diff
@@ -878,3 +883,67 @@ def test_json_diff_scalar_difference():
         b_name="changed",
         context=1,
     ) == ["--- orig\n", "+++ changed\n", "@@ -1 +1 @@\n", '-"string"\n', "+null\n"]
+
+
+def test_merge_collection_metadata_removes_duplicate_links():
+    """A simple test with just one collection that exists on two backends (and is identical on both)."""
+
+    root_url = "https://foo"
+    parent_url = root_url + "/collections"
+    self_url = parent_url + "/nvdi"
+
+    collection_metadata = {
+        "id": "NDVI",
+        "stac_version": "0.9.0",
+        "title": "title of NVDI",
+        "description": "Description of NVDI",
+        "type": "Collection",
+        "links": [
+            {
+                "href": f"{self_url}/about",
+                "rel": "about",
+                "title": "Website describing the collection",
+                "type": "text/html",
+            },
+            {
+                "href": f"{self_url}/somestuff/script.js",
+                "rel": "processing-expression",
+                "title": "Evalscript to do whatever",
+                "type": "application/javascript",
+            },
+            {
+                "href": self_url,
+                "rel": "self",
+            },
+            {
+                "href": parent_url,
+                "rel": "parent",
+            },
+            {
+                "href": root_url,
+                "rel": "root",
+            },
+        ],
+    }
+
+    by_backend = {"backend1": collection_metadata, "backend2": collection_metadata}
```

Comment:
For a clean test of this feature, the self/parent/root urls should start with the upstream back-end domain, instead of a generic "https://foo" in both cases

Reply:
Done. Now it is deriving the URLs from backend1 .